### PR TITLE
[move] Use u256 instead of bigint for parsing

### PR DESCRIPTION
**Pull request by tzakian**
## Description 

Switches to using u256 instead of bigint for parsing. 

## Test plan 

CI + additional tests to make sure parity was preserved.

### CODE REVIEW

File: external-crates/move/crates/move-core-types/src/parsing/parser.rs [hunk 1]
```diff
@@ -447,20 +446,22 @@ pub fn parse_u256(s: &str) -> Result<(U256, NumberFormat), U256FromStrError> {
...
 
 // Parse an address from a decimal or hex encoding
-pub fn parse_address_number(s: &str) -> Option<([u8; AccountAddress::LENGTH], NumberFormat)> {
+pub fn parse_address_number(s: &str) -> Option<(AccountAddress, NumberFormat)> {
     let (txt, base) = determine_num_text_and_base(s);
-    let parsed = BigUint::parse_bytes(
-        txt.as_bytes(),
+    let max_len = match base {
+        NumberFormat::Hex => AccountAddress::LENGTH * 2,
+        NumberFormat::Decimal => 78,
```
Comment (tnowacki) [inline external-crates/move/crates/move-core-types/src/parsing/parser.rs:453]: Where is the 78 coming from? Is it possible to express that in terms of `AccountAddress::LENGTH`?

Comment (tzakian) [reply external-crates/move/crates/move-core-types/src/parsing/parser.rs:453]: Yea! I can do that. It's based off of this: https://stackoverflow.com/questions/43787672/the-max-number-of-digits-in-an-int-based-on-number-of-bits

So would be `241 * AccountAddress::LENGTH / 100 + 1

Comment (tnowacki) [reply external-crates/move/crates/move-core-types/src/parsing/parser.rs:453]: Wowza. Yeah I would definitely comment the shit out of that number lol

Comment (tzakian) [reply external-crates/move/crates/move-core-types/src/parsing/parser.rs:453]: Updated it to be a const + I put a link to the stack overflow article on it as that did a much better job than any explanation I can give.

Comment (tzakian) [reply external-crates/move/crates/move-core-types/src/parsing/parser.rs:453]: I'm reconsidering this move to u256 here altogether -- in particular `_`s in addresses are going to mess this change all up. So will need to look a bit more at it and will land outside of this stack.

**Comment by tzakian** [comment]
Updated so we handle underscores as well now (in integers) + some minor updates to promote parity between the value tokenizer and type tokenizer.

**Review by tnowacki — APPROVED** [review]

**Pull request merged**